specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,940 @@
+winrt::Windows::Foundation::IAsyncOperation<Windows::Foundation::Collections::IVectorView<ToastNotification> > GetToastListASync()
+{
+    auto toastHistory = ToastNotificationManager::History();
+    auto toasts = co_await toastHistory.GetAllAsync();
+    co_return toasts;
+}
+
+// Remove a prior toast using tag and group
+winrt::Windows::Foundation::IAsyncAction RemoveToast(const winrt::hstring& tag, const winrt::hstring& group)
+{
+    auto toastHistory = ToastNotificationManager::History();
+    co_await toastHistory.RemoveWithTagGroupAsync(tag, group);
+}
+```
+
+## Toast Progress Updates
+
+Sometimes a developer would like to show progress bar related updates in a toast:
+
+![Toast Progress](https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/images/toast-progressbar-annotated.png)
+
+To accomplish that, the developer will need to use the ToastProgressData construct.
+
+```cpp
+// Send first Toast Progress Update
+void SendUpdatableToastWithProgress()
+{
+    winrt::hstring tag = L"weekly-playlist";
+    winrt::hstring group = L"downloads";
+
+    const winrt::hstring payload = L"<toast launch=\"action = viewDownload &amp; downloadId = 9438108\">"
+                                L"<visual>"
+                                    L"<binding template = \"ToastGeneric\">"
+                                        L"<text>Downloading this week's new music...</text>"
+                                        L"<progress"
+                                            L"title = \"{progressTitle}\""
+                                            L"value = \"{progressValue}\""
+                                            L"valueStringOverride = \"{progressValueString}\""
+                                            L"status = \"{progressStatus}\" />"
+                                    L"</binding>"
+                                L"</visual>"
+                            L"</toast>";
+
+    winrt::Windows::Data::Xml::Dom::XmlDocument doc = winrt::make<winrt::Windows::Data::Xml::Dom::XmlDocument>();
+    doc.LoadXml(payload);
+    ToastNotification toast = winrt::make<ToastNotification>(doc);
+    toast.Tag(tag);
+    toast.Group(group);
+
+    // Assign initial values for first toast progress UI
+    ToastProgressData data = winrt::make<ToastProgressData>();
+    data.Title(L"Weekly playlist"); // Binds to {progressTitle} in xml payload
+    data.Value(0.6); // Binds to {progressValue} in xml payload
+    data.ValueStringOverride(L"15/26 songs"); // Binds to {progressValueString} in xml payload
+    data.Status(L"Downloading..."); // Binds to {progressStatus} in xml payload
+    data.SequenceNumber(1); // The sequence number has to be incremental for the platform to render progress
+
+    toast.ProgressData(data);
+    ToastNotificationManager::ShowToast(toast);
+}
+
+// Send subsequent progress updates
+winrt::Windows::Foundation::IAsyncAction UpdateProgressAsync()
+{
+    winrt::hstring tag = L"weekly-playlist";
```

Comment:
Maybe we should advice the devs to have such strings as member values? Otherwise, there is a chance to get it wrong.

Reply:
I grow weary of that. From my point of view, its purely an implementation detail. I would argue that they should be arguments to this class but i specified it as a local declaration to get the point across on purpose.